### PR TITLE
[WIP] Preneeds attachments

### PR DESCRIPTION
**Pull request by saneshark**
TODO:

- merge Marc's PR first.
- upmerge with that and add controllers / routes and request specs
- as part of this PR or separate one, handle attachments and virus check similar to Rainbows, vs RT.

### CODE REVIEW

File: lib/preneeds/service.rb [hunk 1]
```diff
@@ -63,5 +87,46 @@ def receive_pre_need_application(params)
...
+
+      soap_part = Mail::Part.new do
+        content_type 'text/xml; charset="utf-8"'
+        body soap.body
+      end
+
+      multipart.add_part(soap_part)
+
+      multipart.attachments[attachment.file.original_filename] = {
+        mime_type: 'application/pdf',
```
Comment (patrickvinograd) [inline lib/preneeds/service.rb]: what guarantees/enforcement do we have that the file is actually a PDF?

File: app/models/preneeds/attachment_type.rb [hunk 2]
```diff
@@ -1,19 +1,9 @@
-# frozen_string_literal: true
+# frozen_string_literal: true'
```
Comment (patrickvinograd) [inline app/models/preneeds/attachment_type.rb]: stray single quote

File: lib/preneeds/middleware/response/clean_response.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@ module Middleware
     module Response
       class CleanResponse < Faraday::Response::Middleware
```
Comment (patrickvinograd) [inline lib/preneeds/middleware/response/clean_response.rb]: We probably also want an html error response middleware - based on pinging endpoints to get connection info, there's likelihood of a HTML error response if something is really fubared (like a 404).

File: lib/preneeds/service.rb [hunk 4]
```diff
@@ -1,50 +1,74 @@
 # frozen_string_literal: true
 require 'common/client/base'
+require 'common/client/concerns/streaming_client'
+require 'net/http/post/multipart'
 
 module Preneeds
   class Service < Common::Client::Base
+    include Common::Client::StreamingClient
```
Comment (saneshark) [inline lib/preneeds/service.rb]: yea, you're right -- i was experimenting with something and forgot to remove this.

File: lib/preneeds/service.rb (hunk 1 above)
Comment (saneshark) [inline lib/preneeds/service.rb]: well the only files that the host allows is 'application/pdf' if they provide anything that doesn't match it should be an error, I suppose we could do a file type validation on the attachment model if that's what you're getting at?

File: app/models/preneeds/attachment_type.rb (hunk 2 above)
Comment (saneshark) [inline app/models/preneeds/attachment_type.rb]: good catch.

File: lib/preneeds/middleware/response/clean_response.rb (hunk 3 above)
Comment (saneshark) [inline lib/preneeds/middleware/response/clean_response.rb]: agree, their errors are terrible though -- it comes back as an XML stacktrace, no concept of minor codes, so not exactly sure how to tackle that at this point.

Comment (mphprogrammer) [inline lib/preneeds/middleware/response/clean_response.rb]: Can't we devise our own codes? - I remember someone saying there might be better errors on their part sometime in the future, but until then.

**Comment by mphprogrammer** [comment]
I am also wondering if, successful, should we track that too, along with the application success?

**Comment by saneshark** [comment]
this has been fixed.

**Pull request closed without merging**